pytensor/tensor/slinalg.py

```diff
@@ -908,6 +908,102 @@ def solve_discrete_are(A, B, Q, R, enforce_Q_symmetric=False) -> TensorVariable:
     )
 
 
+def largest_common_dtype(tensors: typing.Sequence[TensorVariable]) -> np.dtype:
+    return ft.reduce(lambda l, r: np.promote_types(l, r), [x.dtype for x in tensors])
+
+
+class BaseBlockDiagonal(Op):
+    __props__ = ("gufunc_signature",)
+
+    def __init__(self, n_inputs):
```

Comment:
Could save `self.n_inputs` and use in `make_node` where we now check `if not matrices` and instead check that we get the promised number of inputs?

You can do the assert > 0 here already